### PR TITLE
fix(curriculum): cat photo app workshop step 11-15

### LINKED ISSUE #56311: Update steps 11-15 of cat photo workshop to use specific assert methods
We recently updated the guidelines for writing assertions here 
https://contribute.freecodecamp.org/how-to-work-on-coding-challenges/#writing-tests

For consistency and matching what is in the docs, we should slowly start to update the tests for the workshops and labs in the new frontend cert.

For this issue, just focus on updating the tests for steps 11-15 of the cat photo app workshop.
We don't want to have a large PR that updates a whole bunch of tests in a bunch of steps and makes it harder to review. 

Here is the link to the workshop:
https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app

The tests should be updated, where appropriate, to use specific methods like `assert.equal`, `assert.exists`, `assert.match`, etc instead of just `assert`.

Here is the complete API reference 
https://www.chaijs.com/api/assert/#method_exists

Just focus on the new frontend cert and don't worry about updating the RWD cat photo app since that will become legacy anyway.

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56311

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698417821794837f14ca275.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Your `p` element should have a nested anchor (`a`) element with the text `cat ph
 
 ```js
 const anchor = $('p > a');
-assert(
+assert.isTrue(
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698417821794837f14ca275.md:28]: please here use two asserts, one that checks if `anchor`has at least one element (assert.isNotEmpty), and then one that check the inner text of the first one (`assert.equal`)

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ Turn the image into a link by surrounding it with necessary element tags. Use `h
 You should have an `img` element with a `src` value of `https://cdn.freecodecamp.org/curriculum/cat-photo-app/relaxing-cat.jpg`. You may have accidentally deleted it.
 
 ```js
-assert(
+assert.isTrue(
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md:27]: please replace this using optional chaining and `assert.equal`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md [hunk 3]
```diff
@@ -34,46 +34,46 @@ assert(
 Your anchor (`a`) element should have an opening tag. Opening tags have this syntax: `<elementName>`.
 
 ```js
-assert(document.querySelectorAll('a').length >= 2);
+assert.isTrue(document.querySelectorAll('a').length >= 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md:37]: here use `assert.isAtLeast`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md [hunk 4]
```diff
@@ -34,46 +34,46 @@ assert(
...
 ```
 
 You are missing a closing (`a`) tag after the image.
 
 ```js
-assert(document.querySelectorAll('a').length === 2);
+assert.equal(document.querySelectorAll('a').length, 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md:43]: please use `assert.lengthOf`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md [hunk 5]
```diff
@@ -34,46 +34,46 @@ assert(
...
 ```
 
 Your anchor (`a`) element should have a closing tag. Closing tags have a `/` just after the `<` character.
 
 ```js
-assert(code.match(/<\/a>/g).length >= 2);
+assert.isTrue(code.match(/<\/a>/g).length >= 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md:49]: please use `assert.isAtLeast`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md [hunk 6]
```diff
@@ -34,46 +34,46 @@ assert(
...
 ```
 
 You should only add one closing anchor (`a`) tag. Please remove any extras.
 
 ```js
-assert(code.match(/<\/a>/g).length === 2);
+assert.isTrue(code.match(/<\/a>/g).length === 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md:55]: `assert.lengthOf`

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
the title of the workshop is cat photo app, please change to that so that it is understandable for what workshop this issue is for

**Comment by gagan-bhullar-tech** [comment]
@ilenia-magoni updated PR.

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698417821794837f14ca275.md [hunk 7]
```diff
@@ -25,9 +25,9 @@ Your `p` element should have a nested anchor (`a`) element with the text `cat ph
 
 ```js
 const anchor = $('p > a');
-assert.isTrue(
-  anchor.length &&
-    anchor[0].innerText.toLowerCase().replace(/\s+/g, ' ') === 'cat photos'
+assert.isNotEmpty(anchor.length)
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698417821794837f14ca275.md:28]:
```suggestion
assert.isNotEmpty(anchor)
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md [hunk 8]
```diff
@@ -24,35 +24,34 @@ Turn the image into a link by surrounding it with necessary element tags. Use `h
...
 ```js
-assert.isTrue(document.querySelectorAll('a').length >= 2);
+assert.isAtLeast(document.querySelectorAll('a').length, 2);
 ```
 
 You are missing a closing (`a`) tag after the image.
 
 ```js
-assert.equal(document.querySelectorAll('a').length, 2);
+assert.lengthOf(document.querySelectorAll('a').length, 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md:42]:
```suggestion
assert.lengthOf(document.querySelectorAll('a'), 2);
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md [hunk 9]
```diff
@@ -24,35 +24,34 @@ Turn the image into a link by surrounding it with necessary element tags. Use `h
...
 ```js
-assert.isTrue(code.match(/<\/a>/g).length >= 2);
+assert.isAtLeast(code.match(/<\/a>/g).length, 2);
 ```
 
 You should only add one closing anchor (`a`) tag. Please remove any extras.
 
 ```js
-assert.isTrue(code.match(/<\/a>/g).length === 2);
+assert.lengthOf(code.match(/<\/a>/g).length, 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698419d8f52cf3891f34853.md:54]:
```suggestion
assert.lengthOf(code.match(/<\/a>/g), 2);
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698417821794837f14ca275.md [hunk 10]
```diff
@@ -25,22 +25,22 @@ Your `p` element should have a nested anchor (`a`) element with the text `cat ph
 
 ```js
 const anchor = $('p > a');
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698417821794837f14ca275.md:27]:
```suggestion
const anchor = document.querySelector('p > a');
```
Version without jQuery.

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698417821794837f14ca275.md [hunk 11]
```diff
@@ -25,22 +25,22 @@ Your `p` element should have a nested anchor (`a`) element with the text `cat ph
...
-assert(
-  anchor.length &&
-    anchor[0].innerText.toLowerCase().replace(/\s+/g, ' ') === 'cat photos'
+assert.isNotEmpty(anchor);
+assert.equal(
+    anchor[0].innerText.toLowerCase().replace(/\s+/g, ' '), 'cat photos'
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698417821794837f14ca275.md:30]:
```suggestion
assert.equal(
    anchor?.innerText?.toLowerCase().replace(/\s+/g, ' '), 'cat photos'
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698410269ace0373ec812b6.md [hunk 12]
```diff
@@ -33,7 +33,7 @@ There should not be `cat photos` text below the `p` element.
 
 ```js
 const pNextSibling = document.querySelector('p').nextSibling;
-assert(
+assert.isTrue(
   pNextSibling.nodeName === '#text' && !/cat\s*photos/.test(pNextSibling.nodeValue) ||
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698410269ace0373ec812b6.md:37]:
```suggestion
  (pNextSibling.nodeName === '#text' && !/cat\s*photos/.test(pNextSibling.nodeValue)) ||
```
A bit explicit, but I think it makes it much clearer.

**Comment by gagan-bhullar-tech** [comment]
@huyenltnguyen thanks for the tip. Will do that from now on.

**Review by gikf — APPROVED** [review]

**Review by majestic-owl448 — APPROVED** [review]
LGTM 👍

**Pull request merged**